tartiflette/directive/directive.py

```diff
@@ -49,12 +46,8 @@ def bake(self, schema: "GraphQLSchema") -> None:
             )
 
     def __call__(self, implementation):
-        if not iscoroutinefunction(implementation.on_field_execution):
-            raise NonAwaitableDirective(
-                "%s is not awaitable" % repr(implementation)
-            )
+        self._implementation = implementation
```

Comment:
Maybe we could keep a validation on known hooks. Loop over all known available hook and check that if the implementation provide an attribute for a known hook name that this attribute is a coroutine function.